mypy/checkexpr.py

```diff
@@ -1143,17 +1144,23 @@ def check_overload_call(self,
                                                  callable_name=callable_name,
                                                  object_type=object_type)
                 if not unioned_errors.is_errors():
-                    # Success! Stop early.
-                    return unioned_result
+                    # Success! But we need to see maybe normal procedure gives a narrower type.
+                    union_success = True
```

Comment:
I think we can simplify this if statement to `union_success = not unioned_errors.is_errors()`. We'd also need to change the comment above to `# Record if we succeeded. Next we need to see if maybe...`